geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -10149,11 +10147,27 @@ protected PartitionedRegionClear getPartitionedRegionClear() {
     return partitionedRegionClear;
   }
 
+  @Override
+  public void endClear(long startTime) {
+    getCachePerfStats().endClear(startTime);
+  }
+
+  @Override
+  public long startClear() {
+    return getCachePerfStats().startClear();
+  }
+
   @Override
   void cmnClearRegion(RegionEventImpl regionEvent, boolean cacheWrite, boolean useRVV) {
     // Synchronized to avoid other threads invoking clear on this vm/node.
-    synchronized (clearLock) {
-      partitionedRegionClear.doClear(regionEvent, cacheWrite);
+    long startTime = 0L;
+    try {
```

Comment:
why is startClear called inside the try? It seems better to have "final long startTime = startClear(); try...".

Reply:
Moved.